continuous-testing/starter-kits/README.md

```diff
@@ -2,6 +2,403 @@
+> Eventually, it will be merged into Yelp/detect-secrets.
+> 
+> At that time, this document will be updated to use the official version of detect-secrets.
+#### Layer 1: Full Scan and Audit (Client-side)
+The first layer initiates a direct scan on the developer's local environment. This is achieved through the `detect-secrets` tool, which scans the entire codebase and outputs a new baseline file containing detected secrets. The developer can then audit this file to view detailed information about detected secrets.
+```mermaid
+sequenceDiagram
+    participant Dev as Developer
+    participant Env as Local Environment
+    participant DS as Detect-Secrets
+    participant File as Baseline File
+    participant Audit as Audit Tool
+
+    Note over Dev,Env: Developer initiates a direct scan for secrets
+    Dev->>+Env: Triggers direct scan
+    Env->>+DS: Requests scan on the codebase
+    DS->>DS: Performs secret scanning
+    DS->>File: Generates new baseline file
+    File->>DS: Acknowledges file creation
+    DS-->>-Env: Returns scan results and new baseline file
+    Env-->>Dev: Presents scan results and new baseline file
+    Note over Dev,File: Developer may audit the new baseline file
+    Dev->>Audit: Initiates audit on the new baseline file
+    Audit->>File: Fetches details from the baseline file
+    File->>Audit: Returns secret details
+    Audit-->>Dev: Presents detailed information of detected secrets
+
+```
+Starter Kit:
+1. Install experimental version of [slim-detect-secrets](https://github.com/NASA-AMMOS/slim-detect-secrets/tree/exp)
+```bash
+pip install git+https://github.com/NASA-AMMOS/slim-detect-secrets.git@exp
+```
+
+2. Scan all local files from current directory and output the result as a baseline file
+```bash
+detect-secrets scan ./ --all-files --exclude-files '.secrets.*' --exclude-files '.git*' > .secrets.baseline
+```
+Here it scans all the local files from current directory, but excludes `.git` directory and `.secrets` baseline files from scanning to reduce false positive. Make modifications when necessary.
+
+3. Check result in the baseline file
+```bash
+cat .secrets.baseline
+```
+> This command is supported in Unix-like system. For Windows, you can use other command, such as `type .secrets.baseline`
+If any secrets are detected, the result will be located at the "results:" of the file.
+
+For example,
+```json
+"results": {
+    "Jenkins/e2e-opera-pcm-develop.sh": [
+    {
+    "type": "AWS Sensitive Information",
+    "filename": "Jenkins/e2e-opera-pcm-develop.sh",
+    "hashed_secret": "a2c61f79dd90429e5284ab45de3318c16756a2f0",
+    "is_verified": false,
+    "line_number": 39
+},
+```
+Only line number is visible through this approach. You can check them manually in the file, or use the following audit tool to conveniently view the actual secret.
+
+4. Analyze results by `audit` tool
```

Comment:
> In addition to your step 4 here, I think you need add some clarification to the user that they never need to run the initial scan again after the audit if they set up Layer 2.
> 
> i.e. no need to run
> 
> ```
> detect-secrets scan ./ --all-files --exclude-files '.secrets.*' --exclude-files '.git*' > .secrets.baseline
> ```

@riverma I am not quite sure what it mean here. Let me clarify how layer 2 works:

1. When Layer 2 gets triggered, it will look for new secrets by comparing the existing baseline file. That's the reason it has to have a baseline file existing locally.

2. Layer 2 does not change the baseline file automatically. For example, if we have an initial baseline file with empty results, then we add a file with secrets and commit it to trigger Layer 2, then Layer 2 can tell us that secret got detected, but the baseline file remains the same. This is the reason we will use Layer 1 to update the baseline file, which was mentioned [here in the document](https://github.com/NASA-AMMOS/slim/blob/d20ee6134a0dc0e0dab11d2d2570e358ef7e4550/continuous-testing/starter-kits/README.md#:~:text=If%20you%20want%20to%20add%20new%20secret%20results%2C%20you%20need%20to%20update%20.secrets.baseline%20file%20by%20re%2Drunning%20the%20scan%20command%20and%20generate%20a%20new%20baseline%20file.)

I will update the document with this fact more explicitly.

Please let me know if you have any other questions. Thank yoU!

Reply:
@perryzjc LGTM! Thanks for the clarification here.